optimade/server/query_params.py

```diff
@@ -1,10 +1,50 @@
 from fastapi import Query
 from pydantic import EmailStr  # pylint: disable=no-name-in-module
-
+from typing import Iterable
 from optimade.server.config import CONFIG
-
-
-class EntryListingQueryParams:
+from warnings import warn
+from optimade.server.mappers import BaseResourceMapper
+from optimade.server.exceptions import BadRequest
+from optimade.server.warnings import UnknownProviderQueryParameter
+
+
+class BaseQueryParams:
```

Comment:
This should be made an abstract class

```suggestion
from abc import ABC


@ABC
class BaseQueryParams:
```